Failure in arm32/arm64 corefx altjit tests

I'm seeing:
```
10:14:48 Running: D:\j\workspace\x64_arm64_alt---734a66f6\_\fx\Tools\dotnetcli\dotnet msbuild /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false /l:BinClashLogger,Tools\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log /p:OSGroup=Windows_NT /p:ConfigurationGroup=Release /p:ArchGroup=x64 /p:BuildPackages=false  /flp:v=normal;LogFile=build-tests.log  /flp2:warningsonly;logfile=msbuild.wrn  /flp3:errorsonly;logfile=msbuild.err  src/tests.builds /p:WithoutCategories="IgnoreForCI;XsltcExeRequired" /p:PreExecutionTestScript=D:\j\workspace\x64_arm64_alt---734a66f6\SetStressModes.bat
10:14:48 
MSBUILD : error MSB1006: Property is not valid.
10:14:48 Switch: XsltcExeRequired
10:14:48 
10:14:48 For switch syntax, type "MSBuild /help"
10:14:49 Command execution failed with exit code 1.
10:14:49 
10:14:49 D:\j\workspace\x64_arm64_alt---734a66f6>exit 1
```

Did something change in the corefx tools (msbuild, etc.) that makes this syntax invalid now?

This syntax is constructed in coreclr: tests\scripts\run-corefx-tests.py

e.g.: https://ci.dot.net/job/dotnet_coreclr/job/master/view/x64_arm64_altjit/job/jitstress/job/x64_arm64_altjit_checked_windows_nt_corefx_baseline/230/consoleFull#-165810470383554902-aff0-4799-9e92-0ada24ce2a06

--- a/tests/scripts/run-corefx-tests.py
+++ b/tests/scripts/run-corefx-tests.py
@@ -361,20 +361,36 @@ def main(args):
 
     # If we're doing altjit testing, then don't run any tests that don't work with altjit.
     if ci_arch is not None and (ci_arch == 'x86_arm_altjit' or ci_arch == 'x64_arm64_altjit'):
-        # The property value we need to specify is a semicolon separated list of two values,
-        # so the two values must be enclosed in double quotes. Without the quotes, msbuild
-        # thinks the item after the semicolon is a different named property. Also, the double
-        # quotes need preceeding backslashes or else run.exe (invoked from build-tests.cmd)
-        # will eat them. They need to be double backslashes so Python preserves the backslashes.
-        without_categories = '/p:WithoutCategories=\\"IgnoreForCI;XsltcExeRequired\\"'
+        # The property value we need to specify for the WithoutCategories property is a semicolon
+        # separated list of two values, so the two values must be enclosed in double quotes, namely:
+        #
+        #  /p:WithoutCategories="IgnoreForCI;XsltcExeRequired"
+        #
+        # Without the quotes, msbuild interprets the semicolon as separating two name/value pairs,
+        # which is incorrect (and causes an error).
+        #
+        # If we pass this on the command-line, it requires an extraordinary number of backslashes
+        # to prevent special Python, dotnet CLI, CMD, and other command-line processing, as the command
+        # filters through batch files, the RUN tool, dotnet CLI, and finally gets to msbuild. To avoid
+        # this, and make it simpler and hopefully more resilient to scripting changes, we create an
+        # msbuild response file with the required text and pass the response file on to msbuild.
+
+        without_categories_filename = os.path.join(fx_root, 'msbuild_commands.rsp')
+        without_categories_string = '/p:WithoutCategories="IgnoreForCI;XsltcExeRequired"'
+        with open(without_categories_filename, "w") as without_categories_file:
+            without_categories_file.write(without_categories_string)
+        without_categories = "-- @%s" % without_categories_filename
+
+        log('Response file %s contents:' % without_categories_filename)
+        log('%s' % without_categories_string)
+        log('[end response file contents]')
     else:
-        without_categories = '/p:WithoutCategories=IgnoreForCI'
+        without_categories = '-- /p:WithoutCategories=IgnoreForCI'
 
     command = ' '.join((
         command,
         config_args,
         '-SkipTests' if no_run_tests else '',
-        '--',
         without_categories
     ))
 

